### PR TITLE
[docs] Update font and styles

### PR DESCRIPTION
**Pull request by jonsamp**
# Why

Updates the design of docs to be more readable, and to better match expo.io. This includes updating the typeface and many styles.

# Changes

- Replaces Maison with Inter.
- Makes the docs full screen-width. I also constrained the content area to 1200px for large screens.
- Updates block quotes with a new design. Removes border-left design and makes it into a rounded box.
- Updates the left sidebar design. Removes the dashed line, adds styles to make titles and subtitles align correctly. Also updates the selected item style (now there's a dot next to the item you're on).
- Updates the scroll bars with new colors. They're now more subtle and round.
- Updates button design for "Try this example on Snack" buttons.
- Updates design of SDK selector under API reference.
- Replace emojis in the platform compatibility table with icons.
- Updates the permalink icon.
- Updates the external link icon.
- Makes border-radii in the navigation bar 4px.
- Fixes a responsive issue, where between two breakpoints it's impossible to see the sidebar list. 

# Screenshot

Group 1
Screen Shot 2020-09-06 at 6 24 15 PM
Screen Shot 2020-09-06 at 6 24 24 PM
Screen Shot 2020-09-06 at 6 26 57 PM

# Test Plan

You should pull this PR and test it! It affects every page of the docs. I'm looking for styles that are now broken that are not broken on production.

I'm also looking for general design feedback!

### CODE REVIEW

**Comment by brentvatne** [comment]
this is a huge improvement to readability of the docs! just comparing the same page side by side my eyes are drawn to the new version of it. well done!

## the subheading here, "Essentials", doesn't look like it's nested under "Bare workflow"

Screen Shot 2020-09-08 at 10 52 32 AM


## some emojis are being rendered differently

Screen Shot 2020-09-08 at 11 00 01 AM

Screen Shot 2020-09-08 at 11 00 12 AM

Screen Shot 2020-09-08 at 11 03 13 AM

Screen Shot 2020-09-08 at 11 03 58 AM

## block quotes feel too similar to code blocks. before we distinguished them more clearly by removing the border except on the left side where there was a thicker border.

Screen Shot 2020-09-08 at 11 01 07 AM

Screen Shot 2020-09-08 at 11 01 15 AM

## linebreak missing, unsure of cause

Screen Shot 2020-09-08 at 11 05 32 AM

Screen Shot 2020-09-08 at 11 06 35 AM

File: docs/components/DocumentationHeader.js [hunk 1]
```diff
@@ -54,39 +53,39 @@ const STYLES_NAV = css`
...
 const STYLES_MOBILE_NAV = css`
   padding: 0px;
   height: 56px;
+  background: ${Constants.expoColors.white};
+  display: none;
 
-  @media screen and (min-width: ${Constants.breakpoints.mobileStrict}) {
-    display: none;
+  @media screen and (max-width: ${Constants.breakpoints.mobile}) {
+    display: block;
```
Comment (tommybru) [inline docs/components/DocumentationHeader.js:74]:
```suggestion
    display: flex;
```
The searchbar on the mobile nav is not vertically centered. If we want to keep `display: block`, I can look at some other ways to refactor.

**Review by brentvatne — CHANGES_REQUESTED** [review]
commented

File: docs/components/base/typography.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { css } from 'react-emotion';
+import * as Constants from '~/common/constants';
+
+export const h1 = css`
+  font-family: ${Constants.fonts.bold};
+  color: ${Constants.colors.black90};
+  font-size: 52px;
```
Comment (tommybru) [inline docs/components/base/typography.js:7]:
```suggestion
  font-size: 42px;
```
52px seemed a bit too large on desktop and mobile -- 42px felt more comfortable

**Review by tommybru — APPROVED** [review]
Changes look great and very sleek to me. I love the new header with these updates. 

Left comments on small fixes, but am excited for this to merge.

**Review by cruzach — CHANGES_REQUESTED** [review]
this is great! 🥇 making the docs full-width really opens up more room for us to add header links on the right for the individual page you're on (in the future, not this PR of course 😅 )

I have just a few suggestions:

- I think the code text size is too large in comparison to the regular paragraph text now, so either the paragraph text should be bumped up, or the code style text brought down:
image
- same for the float-over element text size, it should probably match the regular paragraph text size:
image
- regarding the code blocks & block quotes brent mentioned- I say we leave the codeblocks as they are, and then block quotes should look less "separated" from the text that's surrounding them. Gmail does a really nice job with block quotes, just an indent and a left border:
image

File: docs/components/base/typography.js (hunk 2 above)
Comment (jonsamp) [reply docs/components/base/typography.js:7]: I made the text match the website, which adheres to a font scale. We can scale down this font stack for the docs if you both think it's too large. I'd just need to scale down all the headers, and then consider if we want to scale down the body text (which is at 16px right now). 

Here's two examples in both directions:
Example of large headers: https://reactnative.dev/docs/intro-react
Example of small headers: https://stripe.com/docs/api

To me, the larger sizes are inviting and make it easier to scan the headers. With that, if it's too much let's scale 'em back.  (cc @brentvatne @tommybru )

**Comment by jonsamp** [comment]
@brentvatne  @cruzach:

## Blockquotes
I changed the blockquote to have a border left style, and also to align the text of the callout with the rest of the text. This serves two purposes: 1. the text flows more since it's all left aligned, 2. the breakout is more of a callout since the container is to the left of everything else. On mobile, I made the container of the block quote align with the surrounding text. Here's a screenshot of a code block and a blockquote:

Screen Shot 2020-09-09 at 11 48 03 AM

## Sidebar
To make the left side bar have better visual hierarchy, I aligned the subtitles with the group titles. The hope is that this structure gives a clear hierarchy of the content.

Screen Shot 2020-09-09 at 11 53 17 AM

## Inline code
I changed the inline code style. It's now a little smaller, rounder, and now has a border. I also made it work inside block quotes. You can see both below:

Screen Shot 2020-09-09 at 12 04 51 PM

## Emojis
Inter displays some emojis differently. My theory is that it has some emojis built-in and then displays them as black/white. I wasn't about to figure out how to fix this without modifying the font file itself, which id like to avoid doing since it's bad for the end-user (they'd have to download our custom Inter, even if they had it cached already). So I changed some emojis, like these arrows:

Screen Shot 2020-09-09 at 12 14 56 PM

One emoji that I couldn't get back to normal is the ⚠️ sign:

Screen Shot 2020-09-09 at 12 15 36 PM

## Tips
I also made the popovers have more reasonable text size/spacing:

Screen Shot 2020-09-09 at 12 14 19 PM

## Other things:
- I added a fuller font-stack for all fonts. 
- I changed all our code fonts to our mono font stack.

File: docs/components/base/typography.js (hunk 2 above)
Comment (tommybru) [reply docs/components/base/typography.js:7]: I'd be open to scaling down just the headers and seeing how the docs look. I think the 16px body text works really well. But I find the H1 a little too large to easily scan.

**Comment by cruzach** [comment]
Hm, personally I think the block quotes should be indented so that they are positioned "inside" the surrounding text, rather than aligned with it. This helps to separate the the vital information from notes and little extra bits of info, which is what we usually use block quotes for in the docs

**Comment by brentvatne** [comment]
I like the new blockquote styling but I prefer it without the negative margin :)

<hr />

I noticed that spacing with emoji seems off with inter maybe?

Screen Shot 2020-09-09 at 11 22 24 AM

it looks like there is no space between the emoji and the text that follows

**Review by cruzach — COMMENTED** [review]
I really like these changes, thank you so much @jonsamp !! 

My final nitpick- I'm not sure if this was changed in one of the recent commits or if I just noticed it this last check, but I think the sidebar page titles are a little bit closer together now. This doesn't really matter when looking at the API sidebar at https://docs.expo.io/versions/latest/ , but it's more noticeable at https://docs.expo.io/guides/, and for me makes it harder to read through them / find something I'm looking for

docs

**Review by cruzach — APPROVED** [review]
awesome, that little bump in space made _such_ an impact on readability

For me, this is ready to 🚀

**Review by brentvatne — APPROVED** [review]
:shipit:

**Pull request merged**